src/main/java/org/mamute/auth/LdapSSOFilter.java

```diff
@@ -0,0 +1,39 @@
+package org.mamute.auth;
+
+import javax.enterprise.event.Observes;
+import javax.inject.Inject;
+import javax.servlet.http.HttpServletRequest;
+
+import org.mamute.dao.UserDAO;
+import org.mamute.model.LoggedUser;
+import org.mamute.model.User;
+
+import br.com.caelum.vraptor.environment.Environment;
+import br.com.caelum.vraptor.events.MethodReady;
+
+public class LdapSSOFilter {
+
+	@Inject private Environment env;
+	@Inject private UserDAO users;
+	@Inject private LDAPApi ldap;
+	@Inject private LoggedUser loggedUser;
+	@Inject private HttpServletRequest request;
+	@Inject private Access system;
+
+	public void checkSSO(@Observes MethodReady methodReady) {
+		if (env.supports(LDAPApi.LDAP_SSO) && !loggedUser.isLoggedIn() && request.getRequestURI() != null
+				&& !request.getRequestURI().endsWith("/logout")) {
+			String userName = request.getRemoteUser();
```

Comment:
Let me quote my initial post:

"To enable SSO, add to your [environment].properties (together with all required LDAP configuration):

ldap.sso = true"

The REMOTE_USER MUST come from outside Mamute (and be configured there), as it is a Web Application Container specific thing.